### PR TITLE
[yarpl] Filter operator for Flowable

### PR DESCRIPTION
**Pull request by phoad**
Implementation of Filter operator 
 - http://reactivex.io/documentation/operators/filter.html

  Flowables::range(0, 10)
    ->filter([] (int64_t x) { return x % 2 != 0; })
    ->subscribe(mySubscriber);

and have `mySubscriber` get sent 1, 3, 5, 7, 9, and then get completed.

### CODE REVIEW

**Comment by lexs** [comment]
I don't see any management of request(n) credits anywhere. Wouldn't the operator have to manage that as it will filter out results? If downstream requests 5 but you filter out 4 you'll have to request 4 more from upstream.

File: experimental/yarpl/test/FlowableTest.cpp [hunk 1]
```diff
@@ -21,6 +21,8 @@ class CollectingSubscriber : public Subscriber<T> {
       std::is_copy_constructible<T>::value,
       "CollectingSubscriber needs to copy the value in order to collect it");
 
+  CollectingSubscriber(int64_t requestCount = 100) : requestCount_(requestCount) {}
+
   void onSubscribe(Reference<Subscription> subscription) override {
     Subscriber<T>::onSubscribe(subscription);
     subscription->request(100);
```
Comment (alexmalyshev) [inline experimental/yarpl/test/FlowableTest.cpp]: I believe you want `subscription->request(requestCount_)`

**Comment by alexmalyshev** [comment]
The PR looks good to me, barring the major conflicts from https://github.com/rsocket/rsocket-cpp/pull/427

**Comment by phoad** [comment]
Thanks for the review Alex and Lex.

**Comment by phoad** [comment]
I am unable to Merge the changes. May I get help in here?

**Pull request merged**